heron/uploaders/src/java/com/twitter/heron/uploader/s3/S3Uploader.java

```diff
@@ -100,6 +100,12 @@ public void initialize(Config config) {
     // is specified. If neither was set just use the DefaultAWSCredentialsProviderChain
     // by not specifying a CredentialsProvider.
     if (!Strings.isNullOrEmpty(accessKey) || !Strings.isNullOrEmpty(accessSecret)) {
+
+      if (!Strings.isNullOrEmpty(awsProfile)) {
+        throw new RuntimeException("Supply either provide heron.uploader.s3.access_key, "
```

Comment:
I think what we want to convey her is that the user must specify either a profile or a key/secret, but not both. Also the wording is off a bit. We should either use supply or provide but not both.